### ISSUE TITLE
Support for rules written in typescript

### ISSUE THREAD

**Issue by AbdealiLoKo** [not-an-aardvark/eslint-plugin-rulesdir#7]
I am trying to write my rule in typescript with https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/development/CUSTOM_RULES.md
But this plugin does not detect that

**Comment by trim21** [comment]
Hi, I just find this feature, and I read document about typescript-eslint. So I write a plugin with in ts and it can't be loaded by eslint.

Is seems that plugin is loaded as pure js file. Are you using some configure like nodejs `--loader` and `ts-node` to make it work?

**Comment by lencioni** [comment]
@trim21 We use some code like the following in our eslint config to make TypeScript work:

```js
// Allow importing typescript files.
process.env.TS_NODE_PROJECT = require.resolve('../typescript/tsconfig.scripts.json');
require('ts-node').register({
  transpileOnly: true,
  // This requires @swc/core or @swc/wasm and @swc/helpers to be installed
  transpiler: 'ts-node/transpilers/swc-experimental',
});
```

**Comment by trim21** [comment]
@lencioni Thanks you very much!
